tests/components/weheat/test_sensor.py

```diff
@@ -0,0 +1,121 @@
+"""Tests for the weheat sensor platform."""
+
+from random import randint
+from unittest.mock import Mock
+
+from weheat.abstractions.heat_pump import HeatPump
+
+from homeassistant.components.weheat.const import DOMAIN, SENSOR_DHW_KEY
+from homeassistant.components.weheat.sensor import (
+    WeheatHeatPumpSensor,
+    WeHeatSensorEntityDescription,
+    async_setup_entry,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def test_native_value(hass: HomeAssistant, mock_coordinator) -> None:
+    """Test the native value of the return the property specified by value_fn."""
+    EXPECTED_VALUE = 1.2
+    entity = WeHeatSensorEntityDescription(
+        translation_key="translation_key",
+        key="key_key",
+        value_fn=lambda status: status.my_property,
+    )
+    mock_coordinator.data.my_property = EXPECTED_VALUE
+    sensor = WeheatHeatPumpSensor(mock_coordinator, entity)
+
+    actual_value = sensor.native_value
+
+    assert actual_value == EXPECTED_VALUE
+
+
+async def test_sensor_creation_no_dhw(hass: HomeAssistant, mock_coordinator) -> None:
+    """Test the sensors do not include DHW related sensors when the heat pump does not have DHW."""
+    config_entry = MockConfigEntry(domain=DOMAIN)
+    config_entry.runtime_data = [mock_coordinator]
+    callback = Mock()
+    await async_setup_entry(hass, config_entry, callback)
+
+    assert len(callback.mock_calls) == 1
```

Comment:
We should not create Sensors or coordinators in the tests. We should rather mock out the library and have HA set up the integration and then observe the state machine what the available states are. Check airgradient for some nice examples :)

Reply:
implemented the snapshot comparable to airgradient